src/coprocessor/dag/expr/mod.rs

```diff
@@ -205,6 +205,23 @@ impl Expression {
         }
     }
 
+    fn eval_string_and_decode<'a, 'b: 'a>(
+        &'b self,
+        ctx: &StatementContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, str>>> {
+        let bytes = try_opt!(self.eval_string(ctx, row));
+        let chrst = self.get_tp().get_charset();
```

Comment:
Is the charset always set?

Reply:
I think so, TiDB will set it to `UTF8` by default. See `tidb/expression/builtin_cast.go:1342`.